src/annotations.cr

```diff
@@ -0,0 +1,11 @@
+# This annotations marks methods as deprecated.
```

Comment:
IMO the description should not limit the usage to methods. This annotation is intended for all features that can be annotated (and appear in a public API), even if initial compiler support only implements it for methods. It can already be used on types or constants as well. It just won't have any effect there yet.

Reply:
The behavior of #7653 is what make me change the description of the annotation. Even considering that you prefer the whole description to mention class, constants, macros even if it not handled at all?

Reply:
Yes. The annotation can be used to mark types, constants and macros as deprecated. If this won't show up with #7653, that's a shortcoming of the docs generator, not the annotation.

Reply:
Note this documentation does not say anything about what it *means* to mark something as deprecated.

Reply:
Docs updated (and I also added the ticks when referring the method, that way the doc generator tool will render the link).